start.sh

```diff
@@ -1,2 +1,362 @@
+  if [ ${#NAMELESS_ARGS[*]} -gt 0 ]; then
+    TEST_TARGET=${NAMELESS_ARGS[1]}
+  fi
+  dbg "Func" "parse_arguments" "Exit"
+}
+
+print_parameters() {
+  dbg "= Parameters ($1):"
+  dbg "TEST_TARGET: "$TEST_TARGET
+  dbg "REPORT_PATH: "$REPORT_PATH
+  dbg "SDL_PROCESS_NAME: "$SDL_PROCESS_NAME
+  dbg "SDL_CORE: "$SDL_CORE
+  dbg "SDL_API: "$SDL_API
+  dbg "IS_REMOTE_ENABLED: "$IS_REMOTE_ENABLED
+  dbg "SAVE_SDL_LOG: "$SAVE_SDL_LOG
+}
+
+check_environment() {
+  dbg "Func" "check_environment" "Enter"
+  local dirs=("test_scripts" "test_sets" "files" "user_modules")
+  for dir in ${dirs[@]}; do
+    if [ ! -d "$dir" ]; then
+      echo "Required folder '$dir' is not available"
+      exit 1
+    fi
+  done
+  dbg "Func" "check_environment" "Exit"
+}
+
+check_arguments() {
+  dbg "Func" "check_arguments" "Enter"
+  if [ -z $TEST_TARGET ]; then
+    echo "Test target was not specified"
+    exit 1
+  fi
+  if [ "${TEST_TARGET: -1}" = "/" ]; then
+    TEST_TARGET="${TEST_TARGET:0:-1}"
+  fi
+  if [ -n $JOBS ] && [ -z "${JOBS##*[!0-9]*}" ]; then
+    echo "Invalid number of jobs was specified within --jobs option"
+    exit 1
+  fi
+  if [ ! -d ${ATF_PATH}/modules/configuration/${CONFIG} ]; then
+    echo "Invalid configuration was specified within --config option"
+    exit 1
+  fi
+  if ( [ $FORCE_PARALLELS = true ] || [ $JOBS -gt 1 ] ) && [ -z "$(docker images | grep 'atf_worker')" ]; then
+    echo "Required docker image 'atf_worker' is not available"
+    exit 1
+  fi
+  dbg "Func" "check_arguments" "Exit"
+}
+
+set_param() {
+  local var=$1
+  local param=$2
+  local config=$3
+  local value=$(get_param_from_atf_config ${config} ${param})
+  if [ -z "${!var}" ] && [ -n "$value" ]; then declare -g -- "$var=$value"; fi
+}
+
+build_parameters() {
+  dbg "Func" "build_parameters" "Enter"
+
+  print_parameters "arguments"
+
+  # load parameters from specific config
+  if [ -n "$CONFIG" ]; then
+    local CONFIG_PATH=${ATF_PATH}/modules/configuration/${CONFIG}
+    set_param SDL_PROCESS_NAME "config.SDL" ${CONFIG_PATH}/base_config.lua
+    set_param REPORT_PATH "config.reportPath" ${CONFIG_PATH}/base_config.lua
+    set_param SDL_CORE "config.pathToSDL" ${CONFIG_PATH}/base_config.lua
+    set_param SDL_API "config.pathToSDLInterfaces" ${CONFIG_PATH}/base_config.lua
+    set_param IS_REMOTE_ENABLED "config.remoteConnection.enabled" ${CONFIG_PATH}/connection_config.lua
+    print_parameters "specific config"
+  fi
+
+
+
+  # load parameters from base config
+  local CONFIG_PATH=${ATF_PATH}/modules/configuration
+  set_param SDL_PROCESS_NAME "config.SDL" ${CONFIG_PATH}/base_config.lua
+  set_param REPORT_PATH "config.reportPath" ${CONFIG_PATH}/base_config.lua
+  set_param SDL_CORE "config.pathToSDL" ${CONFIG_PATH}/base_config.lua
+  set_param SDL_API "config.pathToSDLInterfaces" ${CONFIG_PATH}/base_config.lua
+  set_param IS_REMOTE_ENABLED "config.remoteConnection.enabled" ${CONFIG_PATH}/connection_config.lua
+  print_parameters "base config"
```

Comment:
@jacobkeeler There are 2 steps in process of reading parameters from configs:
1. load parameters from specific config
2. load parameters from base config

Please notice trick is in `set_param` function. It sets parameter value only in case if it was not defined before.
So if parameter is defined in `specific` config (step 1) it won't be overwritten by value from `base` config (step 2)